src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs

```diff
@@ -402,12 +440,32 @@ public sealed partial class HandleCollector
         public void Add() { }
         public void Remove() { }
     }
+    public struct HandleRef
+    {
+        public HandleRef(object wrapper, IntPtr handle): this() { }
+        public IntPtr Handle { get; }
+        public object Wrapper { get; }
+        public static explicit operator IntPtr(HandleRef value) { throw null; }
+        public static IntPtr ToIntPtr(HandleRef value) { throw null; }
+    }
     [System.ComponentModel.EditorBrowsableAttribute((System.ComponentModel.EditorBrowsableState)(1))]
     [System.ObsoleteAttribute("ICustomAdapter may be unavailable in future releases.")]
     public partial interface ICustomAdapter
     {
         object GetUnderlyingObject();
     }
+    public interface ICustomFactory
+    {
+        MarshalByRefObject CreateInstance(Type serverType);
+    }
+    public interface ICustomMarshaler
+    {
+        void CleanUpManagedData(object ManagedObj);
+        void CleanUpNativeData(IntPtr pNativeData);
+        int GetNativeDataSize();
+        IntPtr MarshalManagedToNative(object ManagedObj);
```

Comment:
It works like this but if we regenerate this file it will change it. I will add the qualification.